File: CS/Form1.cs
using System;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraScheduler;
using System.Data.Entity;


namespace SchedulerEntityFramework {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        private CarsXtraSchedulingEntities1 carsXtraSchedulingEntities = new CarsXtraSchedulingEntities1();

        public Form1() {
            InitializeComponent();

            schedulerControl1.Start = new DateTime(2017,3,20);
            schedulerControl1.ActiveViewType = SchedulerViewType.FullWeek;
            schedulerControl1.Storage.FetchAppointments += Storage_FetchAppointments;
            
            SetupMappings();
            carsXtraSchedulingEntities.Cars.Load();
            schedulerControl1.Storage.Resources.DataSource = carsXtraSchedulingEntities.Cars.Local;
            
            schedulerControl1.Storage.AppointmentsChanged += schedulerStorage1_AppointmentsChangedInsertedDeleted;
            schedulerControl1.Storage.AppointmentsInserted += schedulerStorage1_AppointmentsChangedInsertedDeleted;
            schedulerControl1.Storage.AppointmentsDeleted += schedulerStorage1_AppointmentsChangedInsertedDeleted;
        }
 
       
        private void schedulerStorage1_AppointmentsChangedInsertedDeleted(object sender, DevExpress.XtraScheduler.PersistentObjectsEventArgs e) {
            carsXtraSchedulingEntities.SaveChanges();
        }
        private void Storage_FetchAppointments(object sender, FetchAppointmentsEventArgs e) {
            DateTime startTime = e.Interval.Start;
            DateTime endTime = e.Interval.End;

            var entities = carsXtraSchedulingEntities.CarSchedulings.Where(entity => (entity.EventType == 1) ||
                (entity.EventType != 1 &&
                (entity.OriginalStartTime >= startTime && entity.OriginalStartTime < endTime) || (entity.OriginalEndTime > startTime && entity.OriginalEndTime <= endTime)));
            
            entities.Load();
            var collection =  carsXtraSchedulingEntities.CarSchedulings.Local.ToBindingList();
            schedulerControl1.Storage.Appointments.DataSource =collection;
            Text = String.Format("{0} appointments are loaded for {1} interval", collection.Count, e.Interval);
        }

        private void SetupMappings() {
            this.schedulerStorage1.Appointments.Mappings.Subject = "Subject";
            this.schedulerStorage1.Appointments.Mappings.Description = "Description";
            this.schedulerStorage1.Appointments.Mappings.Location = "Location";
            this.schedulerStorage1.Appointments.Mappings.Start = "StartTime";
            this.schedulerStorage1.Appointments.Mappings.End = "EndTime";
             
            this.schedulerStorage1.Appointments.Mappings.AllDay = "AllDay";
            this.schedulerStorage1.Appointments.Mappings.Label = "Label";
            this.schedulerStorage1.Appointments.Mappings.Status = "Status";
            this.schedulerStorage1.Appointments.Mappings.Type = "EventType";
            this.schedulerStorage1.Appointments.Mappings.RecurrenceInfo = "RecurrenceInfo";
            this.schedulerStorage1.Appointments.Mappings.ReminderInfo = "ReminderInfo";
            this.schedulerStorage1.Appointments.Mappings.ResourceId = "CarId";
            this.schedulerStorage1.Appointments.Mappings.OriginalOccurrenceStart = "OriginalStartTime";
            this.schedulerStorage1.Appointments.Mappings.OriginalOccurrenceEnd = "OriginalEndTime";
            this.schedulerStorage1.Resources.Mappings.Id = "ID";
            this.schedulerStorage1.Resources.Mappings.Caption = "Model";

        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Linq
Imports DevExpress.XtraScheduler
Imports System.Data.Entity

Namespace SchedulerEntityFramework

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Private carsXtraSchedulingEntities As CarsXtraSchedulingEntities1 = New CarsXtraSchedulingEntities1()

        Public Sub New()
            InitializeComponent()
            schedulerControl1.Start = New DateTime(2017, 3, 20)
            schedulerControl1.ActiveViewType = SchedulerViewType.FullWeek
            AddHandler schedulerControl1.Storage.FetchAppointments, AddressOf Storage_FetchAppointments
            SetupMappings()
            carsXtraSchedulingEntities.Cars.Load()
            schedulerControl1.Storage.Resources.DataSource = carsXtraSchedulingEntities.Cars.Local
            AddHandler schedulerControl1.Storage.AppointmentsChanged, AddressOf schedulerStorage1_AppointmentsChangedInsertedDeleted
            AddHandler schedulerControl1.Storage.AppointmentsInserted, AddressOf schedulerStorage1_AppointmentsChangedInsertedDeleted
            AddHandler schedulerControl1.Storage.AppointmentsDeleted, AddressOf schedulerStorage1_AppointmentsChangedInsertedDeleted
        End Sub

        Private Sub schedulerStorage1_AppointmentsChangedInsertedDeleted(ByVal sender As Object, ByVal e As PersistentObjectsEventArgs)
            carsXtraSchedulingEntities.SaveChanges()
        End Sub

        Private Sub Storage_FetchAppointments(ByVal sender As Object, ByVal e As FetchAppointmentsEventArgs)
            Dim startTime As Date = e.Interval.Start
            Dim endTime As Date = e.Interval.End
            Dim entities = carsXtraSchedulingEntities.CarSchedulings.Where(Function(entity) entity.EventType = 1 OrElse entity.EventType <> 1 AndAlso entity.OriginalStartTime >= startTime AndAlso entity.OriginalStartTime < endTime OrElse entity.OriginalEndTime > startTime AndAlso entity.OriginalEndTime <= endTime)
            entities.Load()
            Dim collection = carsXtraSchedulingEntities.CarSchedulings.Local.ToBindingList()
            schedulerControl1.Storage.Appointments.DataSource = collection
            Text = String.Format("{0} appointments are loaded for {1} interval", collection.Count, e.Interval)
        End Sub

        Private Sub SetupMappings()
            schedulerStorage1.Appointments.Mappings.Subject = "Subject"
            schedulerStorage1.Appointments.Mappings.Description = "Description"
            schedulerStorage1.Appointments.Mappings.Location = "Location"
            schedulerStorage1.Appointments.Mappings.Start = "StartTime"
            schedulerStorage1.Appointments.Mappings.End = "EndTime"
            schedulerStorage1.Appointments.Mappings.AllDay = "AllDay"
            schedulerStorage1.Appointments.Mappings.Label = "Label"
            schedulerStorage1.Appointments.Mappings.Status = "Status"
            schedulerStorage1.Appointments.Mappings.Type = "EventType"
            schedulerStorage1.Appointments.Mappings.RecurrenceInfo = "RecurrenceInfo"
            schedulerStorage1.Appointments.Mappings.ReminderInfo = "ReminderInfo"
            schedulerStorage1.Appointments.Mappings.ResourceId = "CarId"
            schedulerStorage1.Appointments.Mappings.OriginalOccurrenceStart = "OriginalStartTime"
            schedulerStorage1.Appointments.Mappings.OriginalOccurrenceEnd = "OriginalEndTime"
            schedulerStorage1.Resources.Mappings.Id = "ID"
            schedulerStorage1.Resources.Mappings.Caption = "Model"
        End Sub
    End Class
End Namespace
